ipld/amt/fuzz/fuzz_targets/equivalence.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#![no_main]
+use arbitrary::Arbitrary;
+use fvm_ipld_amt::{Amt, MAX_INDEX};
+use libfuzzer_sys::fuzz_target;
+use cid::Cid;
+
+
+#[derive(Debug, Arbitrary)]
+struct Operation {
+    idx: u64,
+    method: Method,
+}
+
+#[derive(Debug, Arbitrary)]
+enum Method {
+    Insert(u64),
+    Remove,
+    Get,
+}
+
+fn execute(ops: Vec<Operation>) -> (Cid, ahash::AHashMap<u64, u64>) {
+    let db = fvm_shared::blockstore::MemoryBlockstore::default();
+    let mut amt = Amt::new(&db);
+    let mut elements = ahash::AHashMap::new();
+
+    for (i, Operation { idx, method }) in ops.into_iter().enumerate() {
+        if idx > MAX_INDEX {
+            continue;
+        }
+
+        match method {
+            Method::Insert(v) => {
+                elements.insert(idx, v);
+                amt.set(idx, v).unwrap();
+            }
+            Method::Remove => {
+                let el = elements.remove(&idx);
+                let amt_deleted = amt.delete(idx).unwrap();
+                assert_eq!(amt_deleted, el, "step {}", i);
+            }
+            Method::Get => {
+                let ev = elements.get(&idx);
+                let av = amt.get(idx).unwrap();
+                assert_eq!(av, ev, "step {}", i);
+            }
+        }
+    }
+    (amt.flush().unwrap(), elements)
+}
+
+fuzz_target!(|ops: Vec<Operation>| {
```

Comment:
nit: mind documenting what this fuzz target does?

Reply:
i.e. Inserts all elements we know should be existing in the AMT directly into a new AMT, and verifies that the original AMT matches the insert-only AMT.